src/frontend/apps/impress/src/components/dropdown-menu/hook/useDropdownFocusManagement.ts

```diff
@@ -0,0 +1,99 @@
+import { useEffect } from 'react';
+
+interface UseDropdownFocusManagementProps {
+  isOpen: boolean;
+  docId: string;
+  actionsRef?: React.RefObject<HTMLDivElement>;
+}
+
+export const useDropdownFocusManagement = ({
+  isOpen,
+  docId,
+  actionsRef,
+}: UseDropdownFocusManagementProps) => {
+  // Focus management for dropdown menu opening
+  useEffect(() => {
+    if (!isOpen) {
+      return;
+    }
+
+    const timer = setTimeout(() => {
+      // Try to find menu in actions container first
+      const menuElement = actionsRef?.current
+        ?.closest('.--docs--doc-tree-item-actions')
```

Comment:
In `src/components` are only generic components, if the component is specific to a feature, it should be move in it. In this case, if these components are specific to the tree feature, it should be moved in it.